pre_commit/languages/python.py

```diff
@@ -59,10 +58,16 @@ def _norm(path):
     # virtualenv v -ppython2: v/bin/python -> python2
     # virtualenv v -ppython2.7: v/bin/python -> python2.7
     # virtualenv v -ppypy: v/bin/python -> v/bin/pypy
-    for path in {sys.executable, os.path.realpath(sys.executable)}:
+    for path in (sys.executable, os.path.realpath(sys.executable)):
         exe = _norm(path)
         if exe:
             return exe
+    return None
+
+
+def _get_default_version():  # pragma: no cover (platform dependent)
+    # First attempt from `sys.executable` (or the realpath)
+    exe = _find_by_sys_executable()
```

Comment:
this is broken, the value is immediately discarded

Reply:
Wow. Nice "little detail" crept in after refactoring that out to a function for tests.